rocket_packet.c

```diff
@@ -88,17 +91,63 @@ unsigned int serialize_command_packet(CommandPacket* pkt, uint8_t* dst) {
 unsigned int unpack_command_packet(CommandPacket* pkt, uint8_t* src) {
 	unsigned int offset = 0;
 
-	memcpy((void *) &pkt->start_char, src + offset, 2);
-	offset += 2;
+	memcpy((void *) &pkt->start_short, src + offset, sizeof(pkt->start_short));
+	offset += sizeof(pkt->start_short);
+
+	memcpy((void *) &pkt->id, src + offset, sizeof(pkt->id));
+	offset += sizeof(pkt->id);
+
+	memcpy((void *) &pkt->function, src + offset, sizeof(pkt->function));
+	offset += sizeof(pkt->function);
+
+	memcpy((void *) &pkt->arg, src + offset, sizeof(pkt->arg));
+	offset += sizeof(pkt->arg);
+
+	memcpy((void *) &pkt->crc, src + offset, sizeof(pkt->crc));
+	offset += sizeof(pkt->crc);
+
+	return offset;
+}
+
+unsigned int serialize_ack_packet(AckPacket* pkt, uint8_t* dst) {
+	unsigned int offset = 0;
+	
+	memcpy(dst + offset, (void *) &pkt->start_short, sizeof(pkt->start_short));
+	offset += sizeof(pkt->start_short);
+
+	memcpy(dst + offset, (void *) &pkt->id, sizeof(pkt->id));
+	offset += sizeof(pkt->id);
+	
+	memcpy(dst + offset, (void *) &pkt->ack, sizeof(pkt->ack));
+	offset += sizeof(pkt->ack);
+	
+	memcpy(dst + offset, (void *) &pkt->nack, sizeof(pkt->nack));
+	offset += sizeof(pkt->nack);
+	
+	memcpy(dst + offset, (void *) &pkt->crc, sizeof(pkt->crc));
+	offset += sizeof(pkt->crc);
+	
+	return offset;
+}
+
+
+unsigned int unpack_ack_packet(AckPacket* pkt, uint8_t* src) {
+	unsigned int offset = 0;
+
+	memcpy((void *) &pkt->start_short, src + offset, sizeof(pkt->start_short));
+	offset += sizeof(pkt->start_short);
+
+	memcpy((void *) &pkt->id, src + offset, sizeof(pkt->id));
+	offset += sizeof(pkt->id);
 
-	memcpy((void *) &pkt->function, src + offset, 1);
-	offset += 1;
+	memcpy((void *) &pkt->ack, src + offset, sizeof(pkt->ack));
+	offset += sizeof(pkt->ack);
 
-	memcpy((void *) &pkt->arg, src + offset, 1);
-	offset += 1;
+	memcpy((void *) &pkt->nack, src + offset, sizeof(pkt->nack));
+	offset += sizeof(pkt->nack);
```

Comment:
je pense que le ACK/NACK devrait être juste un seul field de la struct, soit il est à 0 (ack) ou 0xff (nack)
ça va être plus facile à gèrer de juste faire if (pkt.ack == 0) {...}
je pense

Reply:
Should we remove NACK part?